### ISSUE TITLE
Consolidate JSON from several Matrix Jobs into single report

### ISSUE THREAD

**Issue by SteveDonie** [jenkinsci/cucumber-reports-plugin#109]
We are using the Matrix job plugin (https://wiki.jenkins-ci.org/display/JENKINS/Matrix+Project+Plugin) to run our cucumber tests - we have quite a few, and running them all serially was taking a very long time. Splitting them up into multiple jobs that run in parallel has helped quite a bit, but now we don't have one report to look at, but many. I know very little about how plugins are implemented, but I am willing to contribute if someone can point me in the right direction. 

Right now, we have a 5x5 matrix of jobs, but sparsen the matrix so we only have 5 jobs total. Here is a screenshot of the 'parent job' page - note that it has no Cucumber Reports link:
![matrixjob1](https://cloud.githubusercontent.com/assets/605418/21811183/594b2f94-d714-11e6-8443-ce8c44bae7f4.png)

And here is one of the 'child jobs' pages, with its cucumber report link. 
![matrixjob2](https://cloud.githubusercontent.com/assets/605418/21811255/a7cd080e-d714-11e6-9061-dc693ceed87b.png)

The JUnit report plugin is able to consolidate the results from matrix jobs like this, so hopefully this is possible.

**Comment by damianszczepanik** [comment]
Hmmm, that is the problem because even you can hardcode some path to have files from different jobs, the paths are relative now. Relative to current workspace.

So my suggestion is to copy all files you need to one location and then run report. Does it make sense?

**Comment by SteveDonie** [comment]
Makes sense, but I am not sure how to control when the report gets run. Right now, I have added the Cucumber-JVM-Reports step to the post-build Actions section of my Matrix job, and it appears that the reports 'run' at the end of each child job. I don't see how to get the matrix job to let me do all the copying and then run the report once all the child jobs have finished. Guess I'll have to start digging into source code to see how the JUnit reports work and then see if I can make the cucumber reports work in a similar manner.

**Comment by blaukvik** [comment]
Any progress on this ?

**Comment by ajmera25** [comment]
Is any one working on this? Thanks

**Comment by tchiaspko** [comment]
Any progress on this?

**Comment by doctau** [comment]
I am doing this by archiving JSOn output as build artifacts in the child jobs, and then copying artifacts in the parent job and running the cucumber plugin there. The only issue is ensuring you have enough memory to process all the results.
